src/components/PageResultsNew/TestResultsTable.tsx

```diff
@@ -0,0 +1,354 @@
+                <Flex
+                    alignSelf={{ default: 'alignSelfCenter' }}
+                    className="pf-u-ml-auto"
+                    spaceItems={{
+                        default: 'spaceItemsNone',
+                    }}
+                >
+                    {props.isWaivable && (
+                        <Button
+                            icon={<ThumbsUpIcon />}
+                            onClick={(event) => {
+                                alert(
+                                    'Not implemented yet. This will show the waiving form.',
+                                );
+                                event.stopPropagation();
+                            }}
+                            variant="link"
+                        >
+                            Waive
+                        </Button>
+                    )}
+                    <Button
+                        icon={<SyncAltIcon />}
+                        onClick={(event) => {
+                            alert(
+                                'Not implemented yet. This will rerun the test.',
+                            );
+                            event.stopPropagation();
+                        }}
+                        variant="link"
+                    >
+                        Rerun
+                    </Button>
+                    <Button
+                        icon={<BookIcon />}
+                        onClick={(event) => {
+                            alert(
+                                'Not implemented yet. This will take you to the CI documentation.',
+                            );
+                            event.stopPropagation();
+                        }}
+                        variant="link"
+                    >
+                        Documentation
+                    </Button>
+                    <Button
+                        icon={<LinkIcon />}
+                        onClick={(event) => {
+                            alert(
+                                'Not implemented yet. This will be a permalink to the test results.',
+                            );
+                            event.stopPropagation();
+                        }}
+                        variant="link"
+                    >
+                        Link
+                    </Button>
+                </Flex>
+            </Flex>
+        </Flex>
+    );
+}
+
+export interface TestResultsTableProps {
+    onSelect?(key: string | undefined): void;
+    tests: CiTest[];
+}
+
+export function TestResultsTable(props: TestResultsTableProps) {
+    const { tests } = props;
+    const selectedTest = useContext(SelectedTestContext);
+
+    // TODO: Handle the empty case, i.e. when there are no (known) tests.
+
+    const failedRequiredRows = tests
+        .filter(
+            ({ required, status }) =>
+                required && (status === 'error' || status === 'failed'),
+        )
+        .map((row) => (
```

Comment:
I would add here a aux constant: testsRequiredUnsatisfied or something  similar.
And map over it in next.

Reply:
Yeah, I'm definitely going to refactor this later on. I'm leaving this as is because it's a bit simpler to play with and move around.